licenses/protobuf-java-BSD-3-clause

```diff
@@ -0,0 +1,34 @@
+Copyright 2023, the original author or authors.
```

Comment:
Google does not have a BSD-3-lcause license in [their repo](https://github.com/protocolbuffers/protobuf/tree/main/java) but it seems [this library is still under that license](https://github.com/protocolbuffers/protobuf/blob/main/java/pom.xml#L44-L45). So I built this file from the license template and picked 2023 as the date.